### DISCUSSION TITLE
Creating a Pull Request - LGMSOC Participant(Muskan)

### DISCUSSION THREAD

**Discussion by prathimacode-hub** [prathimacode-hub/Awesome_Python_Scripts#209] [Q&A]
I shall share you the easiest way to creat a pull request without any hassles especially for beginners to open source and github.

**Comment by muskantayal123** [comment]
@prathimacode-hub Yes Please share it mam.

**Comment by prathimacode-hub** [comment]
Process to Follow (without command line confusion)👍

Fork the repository. You are now in your profile repository.
You can see a "Add File" option in the local repository
You can add files or folders using the 2options given
Create new file option helps in creating both file and sub repository within this repository.
Example : You want to add up your "Snake Game" Script to the repository
Go to "PyGamesScripts" repository. Select "Create a new file" option,
It opens up like this - /Awesome_Python_Scripts/PyGamesScripts
To add up your repository and file you do :
/Awesome_Python_Scripts/PyGamesScripts/Snake Game/README.md
This add up would not only add 'Snake Game' as repository but also adds a empty 'README.md' file.
You can add your README content in 'README.md file' and commit the changes
Now you have created a repository of your desired script, you are yet to add your python program and related files.
From the very same "Add File" option, select "Upload files" and add your python program from your local drive of your system.
Make sure the script file name is snake_game.py before you upload the file to this repository.
Now, we needed to add images and screenshots of your project, so we again use " Create a New File" Option  you do
/Awesome_Python_Scripts/PyGamesScripts/Snake Game/Images/README.md and commit it
You're now in "Images" folder. Add the relevant images and screenshots you want to add up in your README.md file of your project using "Add File" Option from local drive of your system
Get back to your project repository of Snake Game and using "Create a New File" Option you do
/Awesome_Python_Scripts/PyGamesScripts/Snake Game/requirements.txt, add the imported libraries used in your program in this file and commit it
We haven't added our output screenshots or images in "README.md" of your project. Open up your README file. There you could see a "pencil button" that is to edit the file. You can add up the image links you had stored in "Images" folder in README file
If we remember, we had created a README.md file in "Images" folder as well to get it registered as a repository. 
You can now delete that file using "delete option that is trash button". And you are done.
Finally, You have successfully added up all your required files needed.
Now select "Contribute" option above and select Open Pull request, and add a defined title of your project, a proper commit message in a description using the PR template given.
Finally, "Create a Pull Request" option to synchronize your fork to the committed repository. That's it. 
Voila, you did it. Congrats, on learning how to create a PR in a easier way. No much fuss around to deal with.
PR can be created without the use of command line terminal, but preferrably that way to show professionalism.
Contribution is all that matters, so start contributing and feel free to do more PR in future.

@muskantayal123 , I like your confidence and enthusiasm to learn. Keep going.
If you found this helpful and easier to try, you can do create the PR of the same script with required changes I have mentioned in this comment.

Looking forward to merge the upcoming PR from you. Thank You.

**Comment by muskantayal123** [comment]
@prathimacode-hub Thank You soo much  mam . Thank you for this detailed explanation  and i'll try ma'am . I hope i am not bothering you. If i'll face any issue i'll get back to you.

**Comment by prathimacode-hub** [comment]
Nope, not at all. Any issue, kindly get back to me. There was one participant earlier, who faced this same issue. This solution helped her. Hope it would help you too. I can wait to see you finish your PR succesfully. @muskantayal123

**Comment by muskantayal123** [comment]
@prathimacode-hub I am not able to generate a Pull Request . I have followed all the steps till contribute. when i select contribute and open pull request but it was not showing to make a pull request.  I am a bit confused how to proceed from this step.
![Screenshot (1622)](https://user-images.githubusercontent.com/62842695/122013944-4d669600-cddc-11eb-9ad2-38624041cb6c.png)

**Comment by prathimacode-hub** [comment]
That's because your PR is already created in this repo. So it isn't showing. I shall close your PR. You follow the same steps again. You would get to see an option to create a pull request. @muskantayal123

**Comment by muskantayal123** [comment]
@prathimacode-hub From starting i have to repeat all the steps? or else if you close then i can directly contribute only.

**Comment by prathimacode-hub** [comment]
I had closed the PR. You have to repeat from the start for smoother process of contribution. @muskantayal123
